### PR TITLE
Extend giflib harness: Cover more functions and add protobuf specification

### PR DESCRIPTION
**Pull request by viniul**
CC @kcc 
This PR, written by @bshastry and me, extends the giflib harness in two ways:
-> Extend the tested functionality in order to cover more of giflib's code.
-> Add a protobuf specification and a protobuf fuzzer (see [here](https://github.com/google/fuzzer-test-suite/blob/master/tutorial/structure-aware-fuzzing.md)) for giflib. This might be particularly interesting to compare non-proto vs. protobuf fuzzers.

### CODE REVIEW

**Comment by bshastry** [comment]
I am okay with this contribution.

File: projects/giflib/dgif_fuzz_common.cc [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	if (GifFile == NULL){
+		free(gifData);
+		return 0;
+	}
+	if(DGifSlurp(GifFile) != GIF_OK){
+		DGifCloseFile(GifFile, &Error);
+		free(gifData);
+		return 0;
+	}
+    GraphicsControlBlock gcb;
```
Comment (kcc) [inline projects/giflib/dgif_fuzz_common.cc]: is something wrong with formatting here?

**Comment by viniul** [comment]
I pushed two new commits that fix the formatting inconsistencies and fix the build error. Regarding the CLA: Unfortunately, I can't find an email that is not signed at the moment. The googlebot comment states that "everyone that needs to sign a CLA [...] have done so", but we need confirmation from the co-authors of this pull request, which @bshastry (the only other co-author) has given. Anything else needed from our side I am overlooking?

**Comment by kcc** [comment]
Not sure what to do with the CLA. Others? 

egif_target seems to not produce any coverage. Is that expected?
```
build/out/giflib/egif_target 
INFO: Seed: 3265009299
INFO: Loaded 1 modules   (390 inline 8-bit counters): 390 [0x7d75e0, 0x7d7766), 
INFO: Loaded 1 PC tables (390 PCs): 390 [0x5938c8,0x595128), 
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
INFO: A corpus is not provided, starting from an empty corpus
#2	INITED cov: 18 ft: 19 corp: 1/1b exec/s: 0 rss: 33Mb
#131072	pulse  cov: 18 ft: 19 corp: 1/1b lim: 1300 exec/s: 65536 rss: 379Mb
#262144	pulse  cov: 18 ft: 19 corp: 1/1b lim: 2611 exec/s: 65536 rss: 397Mb
#524288	pulse  cov: 18 ft: 19 corp: 1/1b lim: 4096 exec/s: 58254 rss: 414Mb
```

**Comment by bshastry** [comment]
> egif_target seems to not produce any coverage. Is that expected?

Yes, we need to improve coverage there. We can do it in follow up PR or would you prefer it being done in this PR?

**Comment by bshastry** [comment]
Hello everyone,

Is there anything I can help with to fix blockers for merging this PR?

Thank you.

**Comment by Dor1s** [comment]
CLA is good, there are two emails used, and both of them have CLA signed for:

```bash
$ git log -n 29 | egrep -i author | sort | uniq
Author: Bhargava Shastry <bshas3@gmail.com>
Author: Vincent Ulitzsch <vincent@srlabs.de>
```

The CLA bot is not capable to verify that and we have to do it manually -- done.

**Pull request merged**